spec/models/service_ansible_playbook_spec.rb

```diff
@@ -0,0 +1,92 @@
+describe(ServiceAnsiblePlaybook) do
+  let(:tower_job)      { FactoryGirl.create(:ansible_tower_job) }
+  let(:tower_job_temp) { FactoryGirl.create(:ansible_configuration_script) }
+  let(:basic_service)  { FactoryGirl.create(:service_ansible_playbook, :options => dialog_options) }
+
+  let(:loaded_service) do
+    service_template = FactoryGirl.create(:service_template)
+    service_template.resource_actions.build(:action => 'Provision', :configuration_template => tower_job_temp)
+    service_template.save!
+    FactoryGirl.create(:service_ansible_playbook, :options => provision_options, :service_template => service_template)
+  end
+
+  let(:executed_service) do
+    basic_service.tap do |service|
+      allow(service).to receive(:job).with('Provision').and_return(tower_job)
+    end
+  end
+
+  let(:dialog_options) do
+    {
+      :dialog => {
+        'dialog_hosts'      => 'host1;host2',
+        'dialog_credential' => 'machine_secret',
+        'dialog_param_var1' => 'value1',
+        'dialog_param_var2' => 'value2'
+      }
+    }
+  end
+
+  let(:override_options) do
+    {
+      :hosts      => 'host3',
+      :extra_vars => { 'var1' => 'new_val1' }
+    }
+  end
+
+  let(:provision_options) { {:provision_job_options => override_options} }
+
+  describe '#preprocess' do
+    it 'prepares options for action' do
+      basic_service.preprocess('Provision', override_options)
+      basic_service.reload
+      expect(basic_service.options[:provision_job_options]).to have_attributes(
+        :hosts      => 'host3',
+        :credential => 'machine_secret',
+        :extra_vars => {'var1' => 'new_val1', 'var2' => 'value2'}
+      )
+    end
+  end
+
+  describe '#execute' do
+    it 'creates an Ansible Tower job' do
+      expect(ManageIQ::Providers::AnsibleTower::AutomationManager::Job)
+        .to receive(:create_job).with(tower_job_temp, provision_options[:provision_job_options]).and_return(tower_job)
+      loaded_service.execute('Provision')
```

Comment:
@bzwei 
Should Provision, Reconfigure, Retirement be defined as a constant in the ServiceTemplate for everyone to use.